### PR TITLE
 PMM-11038 Added kubernetes client

### PR DESCRIPTION
**Pull request by gen1us2k**
PMM-0

Build: SUBMODULES-0

- [ ] Links to other linked pull requests (optional).

### CODE REVIEW

File: managed/services/dbaas/kubernetes/kubernetes.go [hunk 1]
```diff
@@ -54,8 +67,59 @@ func NewIncluster() (*Kubernetes, error) {
...
+			Transport: &http.Transport{
+				MaxIdleConns:    1,
+				IdleConnTimeout: 10 * time.Second,
+			},
+		},
+	}
+}
+
+// New returns new Kubernetes object.
+func (k *Kubernetes) ChangeKubeconfig(ctx context.Context, kubeconfig string) error {
```
Comment (percona-csalguero) [inline managed/services/dbaas/kubernetes/kubernetes.go:108]: Just a minor comment. I would name this method as 'SetKubeConfig'

**Review by BupycHuk — APPROVED** [review]

File: managed/services/dbaas/kubernetes/kubernetes.go [hunk 2]
```diff
@@ -70,3 +134,104 @@ func (k *Kubernetes) GetKubeconfig(ctx context.Context) (string, error) {
...
+
+// GetOperatorVersion parses operator version from operator deployment
+func (k *Kubernetes) GetOperatorVersion(ctx context.Context, name string) (string, error) {
+	k.lock.RLock()
+	defer k.lock.RUnlock()
+	deployment, err := k.client.GetDeployment(ctx, name)
+	if err != nil {
+		return "", err
+	}
+	return strings.Split(deployment.Spec.Template.Spec.Containers[0].Image, ":")[1], nil
```
[the remark this answers is not in the document]
Comment (gen1us2k) [reply managed/services/dbaas/kubernetes/kubernetes.go:215]: Yes. The Cloud team follows this pattern.

[the remark this answers is not in the document]
Comment (percona-csalguero) [reply managed/services/dbaas/kubernetes/kubernetes.go:215]: :+1:

File: managed/services/dbaas/kubernetes/client/client.go [hunk 3]
```diff
@@ -49,27 +63,79 @@ var (
...
+}
 
+func (c *Client) setup() error {
 	namespace := "default"
 	if space := os.Getenv("NAMESPACE"); space != "" {
 		namespace = space
 	}
+	// Set PATH variable to make aws-iam-authenticator executable
+	path := fmt.Sprintf("%s:%s", os.Getenv("PATH"), dbaasToolPath)
+	os.Setenv("PATH", path)
```
Comment (recharte) [inline managed/services/dbaas/kubernetes/client/client.go:125]: @gen1us2k is this needed after all?
https://github.com/percona/pmm/pull/1407#discussion_r1023703215

**Pull request merged**